src/elli/database/refractive_index_info.py

```diff
@@ -167,7 +244,25 @@ def _search(
         else:
             result = self.catalog.loc[self.catalog[subcatalog] == query]
 
-        return result
+        if wavelength_filter is None:
+            return result
+        elif isinstance(wavelength_filter, (int, float)):
+            return result.loc[
+                (result.lower_range <= wavelength_filter)
+                & (result.upper_range >= wavelength_filter)
+            ]
+        elif isinstance(wavelength_filter, list):
```

Comment:
I think it would be nice to also allow a tuple of ints/floats here. Nicest would be any iterable of floats/ints but I think it's not so easy to isinstance check it in python.